torch/csrc/utils/tensor_new.cpp

```diff
@@ -166,7 +166,10 @@ ScalarType infer_scalar_type(PyObject *obj) {
     return numpy_dtype_to_aten(PyArray_TYPE((PyArrayObject*)obj));
   }
   if (PyArray_CheckScalar(obj)) {
-    return numpy_dtype_to_aten(PyArray_TYPE((PyArrayObject*)(PyArray_FromScalar(obj, nullptr))));
+    auto arr = PyArray_FromScalar(obj, nullptr);
```

Comment:
can we just use THPObjectPtr or whatever here so we don't have to do manually memory management?